src/libexpr/primops.cc

```diff
@@ -705,32 +705,52 @@ static RegisterPrimOp primop_genericClosure(PrimOp {
     .args = {"attrset"},
     .arity = 1,
     .doc = R"(
-      Take an *attrset* with values named `startSet` and `operator` in order to
-      return a *list of attrsets* by starting with the `startSet` and recursively
-      applying the `operator` function to each `item`. The *attrsets* in the
-      `startSet` and the *attrsets* produced by `operator` must contain a value
-      named `key` which is comparable. The result is produced by calling `operator`
-      for each `item` with a value for `key` that has not been called yet including
-      newly produced `item`s. The function terminates when no new `item`s are
-      produced. The resulting *list of attrsets* contains only *attrsets* with a
-      unique key. For example,
+      `builtins.genericClosure` computes the transitive closure over an arbitrary relation defined by an `operator` function.
+      It is designed to handle potentially cyclic data structures or any complex language constructs that require iterative processing.
```

Comment:
```suggestion
      `builtins.genericClosure` iteratively computes the transitive closure over an arbitrary relation defined by an `operator` function.
```

I would refrain from saying something is done with some purpose. Too often have I been misled by such statements. Also I don't understand what "any complex language constructs" are supposed to be and prefer to leave it out.

What's left is mysterious enough but probably not wrong.